include/nuttx/serial/tioctl.h

```diff
@@ -198,6 +198,14 @@
 
 #define SER_SWAP_ENABLED   (1 << 0) /* Enable/disable RX/TX swap */
 
+/* Single-wire UART Duplex */
+
+#define TIOCSSINGLEWIREDUPLEX _TIOC(0x0037)  /* Set single-wire duplex mode */
+#define TIOCGSINGLEWIREDUPLEX _TIOC(0x0038)  /* Get single-wire duplex mode */
```

Comment:
why not continue use TIOCSSINGLEWIRE?

Reply:
I've looked at that before, but re-using `TIOCSSINGLEWIRE` requires state tracking whether `SER_SINGLEWIRE_ENABLED` is enabled. Using `TIOCSSINGLEWIREDUPLEX` we can keep it simple and stateless.

Reply:
so, the user need call two ioctl to enable singlewire mode?